include/jlcxx/stl.hpp

```diff
@@ -311,15 +322,28 @@ struct WrapSTLContainer<std::vector> : STLTypeWrapperBase<WrapSTLContainer<std::
     using T = typename WrappedT::value_type;
     wrapped.module().set_override_module(stl_module());
     wrapped.method("cppsize", &WrappedT::size);
-    wrapped.method("resize", [] (WrappedT& v, const cxxint_t s) { v.resize(s); });
+
+    // Although the C++ standard sates potential undefined behavior depending on
+    // if T is DefaultInsertable/MoveInsertable/CopyInsertable, the only hard
+    // type requirement for WrappedT<T>::resize is that if the current size
+    // is less than the target resize, that T must be DefaultConstructable.
+    if constexpr(std::is_default_constructible_v<T>) {
+      wrapped.method("resize", [] (WrappedT& v, const cxxint_t s) { v.resize(s); });
+    }
+
+    // The exposed `append' method uses WrappedT<T>::reserve, which requires
+    // T to be MoveInsertible into WrappedT<T>. If this is not satisfied,
+    // we can simply fall back to the slower non-reserved method.
     wrapped.method("append", [] (WrappedT& v, jlcxx::ArrayRef<T> arr)
     {
       const std::size_t addedlen = arr.size();
-      v.reserve(v.size() + addedlen);
-      for(size_t i = 0; i != addedlen; ++i)
-      {
-        v.push_back(arr[i]);
+      if constexpr(jlcxx::detail::is_move_insertable_v<T, typename WrappedT::allocator_type>) {
+	v.reserve(v.size() + addedlen);
```

Comment:
Small problem with the indentation here.

Reply:
Fixed, and checked other changes for this as well :+1: